service/src/test/java/bio/terra/externalcreds/dataAccess/SshKeyPairDaoTest.java

```diff
@@ -0,0 +1,138 @@
+      verifySshKeyPair(sshKeyTwo, loadedSshKeyTwo.get());
+    }
+
+    @Test
+    void testUpsertTwiceWithDifferentUserId() {
+      var userId = UUID.randomUUID().toString();
+      var externalUserEmail = "bar@monkeyseesmonkeydo.com";
+      var sshKeyOne = createDefaultSshKey();
+      var sshKeyTwo = createSshKey(userId, DEFAULT_KEY_TYPE, externalUserEmail);
+      sshKeyPairDAO.upsertSshKeyPair(sshKeyOne);
+      sshKeyPairDAO.upsertSshKeyPair(sshKeyTwo);
+
+      var loadedKeyOne = sshKeyPairDAO.getSshKeyPair(DEFAULT_USER_ID, DEFAULT_KEY_TYPE);
+      var loadedKeyTwo = sshKeyPairDAO.getSshKeyPair(userId, DEFAULT_KEY_TYPE);
+
+      assertPresent(loadedKeyOne);
+      assertPresent(loadedKeyTwo);
+      verifySshKeyPair(sshKeyOne, loadedKeyOne.get());
+      verifySshKeyPair(sshKeyTwo, loadedKeyTwo.get());
+    }
+
+    @Test
+    void upsertSshKey() {
+      var sshKey = createDefaultSshKey();
+      var storedSshKey = sshKeyPairDAO.upsertSshKeyPair(sshKey);
+
+      verifySshKeyPair(sshKey, storedSshKey);
+    }
+  }
+
+  @Nested
+  class GetSshKeyPair {
+    @Test
+    void testGetSshKeyPairWithoutUserId() {
+      var empty = sshKeyPairDAO.getSshKeyPair("", DEFAULT_KEY_TYPE);
+
+      assertEmpty(empty);
+    }
+
+    @Test
+    void testGetSshKeyPair() {
+      var sshKey = createDefaultSshKey();
+      sshKeyPairDAO.upsertSshKeyPair(sshKey);
+
+      var loadedSshKeyOptional = sshKeyPairDAO.getSshKeyPair(DEFAULT_USER_ID, DEFAULT_KEY_TYPE);
+
+      assertPresent(loadedSshKeyOptional);
+      verifySshKeyPair(sshKey, loadedSshKeyOptional.get());
+    }
+  }
+
+  @Nested
+  class DeleteKeyPair {
+    @Test
+    void testDeleteSshKeyPair() {
+      sshKeyPairDAO.upsertSshKeyPair(createDefaultSshKey());
+
+      assertTrue(sshKeyPairDAO.deleteSshKeyPair(DEFAULT_USER_ID, DEFAULT_KEY_TYPE));
+
+      assertEmpty(sshKeyPairDAO.getSshKeyPair(DEFAULT_USER_ID, DEFAULT_KEY_TYPE));
+      assertFalse(sshKeyPairDAO.deleteSshKeyPair(DEFAULT_USER_ID, DEFAULT_KEY_TYPE));
+    }
+
+    @Test
+    void testDeleteNonExistingSshKeyPair() {
+      assertFalse(sshKeyPairDAO.deleteSshKeyPair(DEFAULT_USER_ID, DEFAULT_KEY_TYPE));
+    }
+
+    @Test
+    void deleteSshKeyPairWithWrongType() {
+      sshKeyPairDAO.upsertSshKeyPair(createDefaultSshKey());
+
+      assertFalse(sshKeyPairDAO.deleteSshKeyPair(DEFAULT_USER_ID, SshKeyPairType.AZURE));
+
+      assertTrue(sshKeyPairDAO.deleteSshKeyPair(DEFAULT_USER_ID, DEFAULT_KEY_TYPE));
+    }
+  }
+
+  private static SshKeyPair createDefaultSshKey() {
+    return createSshKey(DEFAULT_USER_ID, DEFAULT_KEY_TYPE, DEFAULT_EXTERNAL_USER_EMAIL);
+  }
+
+  private static SshKeyPair createSshKey(
+      String userId, SshKeyPairType keyType, String externalUserEmail) {
+    return new SshKeyPair.Builder()
+        .type(keyType)
+        .privateKey(DEFAULT_PRIVATE_KEY)
+        .publicKey(DEFAULT_PUBLIC_KEY)
+        .userId(userId)
+        .externalUserEmail(externalUserEmail)
+        .build();
+  }
```

Comment:
Your tests are too reliant on the `DEFAULT` constants. This makes them more brittle and harder to follow the intent. I think you should replace `createDefaultSshKey` with `createRandomSshKey` and remove those constants which will force the tests to do it right. 

`createSshKey` is also not required and should be replaced with `createRandomSshKey().withXXX`.

This will make the tests clear about what attributes are important.